src/lib.rs

```diff
@@ -1,185 +1,33 @@
 #![doc(html_root_url = "https://marad.github.io/hyper-router/doc/hyper_router")]
 
-//! # Hyper Router
```

Comment:
Why did you remove the documentation from the lib?

Reply:
I don't recall 100% why I removed it, since it's been a while since I looked at this PR. IIRC, I was moving stuff around different modules, and so to make it easier, I removed the docs, and was thinking of waiting for the dust to settle before adding the docs back in, and possibly updating them in light of my additions.

I wonder if it might make sense to close this PR, and start from scratch (or I can salvage bits and pieces from this), maybe you can tell me how you envision this functionality working, and we can go from there, with everything fresh in both our minds? :D

Reply:
Wouldn't it be a waste of your work? This PR looks quite good. Would you approach this differently now?